### PR TITLE
Result: added Access Denied Error + Added check on syscall sys_process_cancel

### PR DESCRIPTION
**Pull request by Keyboard-Slayer**

### CODE REVIEW

File: kernel/tasking/Syscalls.cpp [hunk 1]
```diff
@@ -76,6 +76,11 @@ Result sys_process_cancel(int pid)
 {
     AtomicHolder holder;
 
+    if (pid < 2)
```
Comment (sleepy-monax) [inline kernel/tasking/Syscalls.cpp:79]:
```
Task *task = task_by_id(pid);

if (!task->user)
{
    return ERR_ACCESS_DENIED;
}
```

Comment (Keyboard-Slayer) [reply kernel/tasking/Syscalls.cpp:79]: ow

File: kernel/tasking/Syscalls.cpp [hunk 2]
```diff
@@ -78,6 +78,11 @@ Result sys_process_cancel(int pid)
 
     Task *task = task_by_id(pid);
 
+    if (!task->user)
+    {
+        return ERR_ACCESS_DENIED;
+    }
+
     if (task == nullptr)
```
Comment (sleepy-monax) [inline kernel/tasking/Syscalls.cpp:86]: You put the user check before the null check, enjoy your null deref/kernel panic :p

Comment (Keyboard-Slayer) [reply kernel/tasking/Syscalls.cpp:86]: big oof

**Comment by sleepy-monax** [comment]
Awesome, I'm merging that :)

**Pull request merged**